lib/mix/tasks/show_vars.ex

```diff
@@ -2,31 +2,37 @@ defmodule Mix.Tasks.ShowVars do
   @shortdoc "Show the names of the environment variables to use"
 
   @moduledoc """
-  This taks will look at the EnvVar.Config you have defined
+  This taks will look at the config you have defined
   and return the list of environment variables you will need
   to supply to the release at runtime.
   """
 
   use Mix.Task
 
   def run([prefix]) do
-    Mix.Task.run("compile")
-
-    # Dynamic module name avoids compilation warnings.
-    config_module = EnvVar.Config
-
-    if Code.ensure_compiled?(config_module) do
-      config = config_module.config()
-      EnvVar.Provider.show_vars(prefix: prefix, env_map: config)
-    else
-      Mix.raise("""
-      You must include a module called EnvVar.Config and supply
-      a config/0 function that returns the release config.
-      """)
-    end
+    Mix.Project.config()
+    |> Keyword.fetch!(:releases)
+    |> fetch_only_release!()
+    |> Keyword.fetch!(:config_providers)
+    |> find_env_var_provider_config!()
+    |> Keyword.put(:prefix, prefix)
+    |> EnvVar.Provider.show_vars()
```

Comment:
Instead of requiring a hard-coded `EnvVar.Config` module, we fetch the release configuration from `mix.exs`, find the options we gave to the `EnvVar.Provider` provider, and use that to show the vars.

Reply:
Nice!

Reply:
Does this actually work when you remove `Mix.Task.run("compile")`? In my experience it does not compile the file before running Mix and only works if you happened to have compiled it already.

Reply:
Also, I'm good with this approach as long as in our own apps we continue to configure things in the same modules we are using now and which everyone knows.

Reply:
> Also, I'm good with this approach as long as in our own apps we continue to configure things in the same modules we are using now and which everyone knows.

It's going to work with `EnvVar.Config` yeah, it just doesn't **need** to be that anymore. I'd prefer to move to `Digits.EnvVarConfig` instead of the "magic" module name but even then, it's a discussion we don't need to have right now at all (and maybe it will never be worth it) :)